test/Serialization/upgrades/runtests.jl

```diff
@@ -1,38 +1,51 @@
 @testset "Serialization.Upgrades" begin
   @testset "< 0.11.3 Upgrade" begin
     # test loading
-    load(joinpath(@__DIR__, "file_version_less_than_0.11.2.json"))
+    path = joinpath(Main.serialization_upgrade_test_path, "version_0_11_0", "QQPolyRingElem.mrdi")
+    @test load(path) isa QQPolyRingElem
 
     L = ones(QQFieldElem, 15)
     R, x = QQ[:x]
     p = R(L)
-    loaded_p = load(joinpath(@__DIR__, "file_version_less_than_0.11.2.json"); params=R);
+    loaded_p = load(path; params=R);
     @test p == loaded_p
+
+    loaded_container1 = load(joinpath(Main.serialization_upgrade_test_path, "version_0_11_0", "Containers.mrdi"))
+    loaded_container2 = load(joinpath(Main.serialization_upgrade_test_path, "version_0_11_3", "Containers.mrdi"))
+    @test loaded_container1 == loaded_container2
+    @test loaded_container1 == (r = QQFieldElem(1, 2), m = QQFieldElem[1//2 1; 0 1], t = (1, 2, 3))             
   end
 
   @testset "< 0.12.0 Upgrade" begin
     # test loading
-    load(joinpath(@__DIR__, "file_version_less_than_0.12.0.json"));
+    path = joinpath(Main.serialization_upgrade_test_path, "version_0_11_3", "fqPolyRepPolyRingElem.mrdi")
+    @test load(path) isa fqPolyRepPolyRingElem
     
     Zt, t = polynomial_ring(residue_ring(ZZ, 2)[1], :t)
     Fin, d = Nemo.Native.finite_field(t^2 + t + 1)
     Rx, x = Fin[:x]
     p = x^2 + d * x + 1
-    loaded_p =  load(joinpath(@__DIR__, "file_version_less_than_0.12.0.json"); params=Rx);
+    loaded_p = load(path; params=Rx);
     @test p == loaded_p
   end
 
   @testset "< 1.2.0 Upgrade" begin
     # test loading
-    loaded = load(joinpath(@__DIR__, "file_version_less_than_1.2.0.json"));
+    loaded = load(joinpath(Main.serialization_upgrade_test_path, "version_1_1_0", "Dict.mrdi"))
     @test loaded isa Dict
+    @test length(loaded) == 7
   end
 
   @testset "< 1.3.0 Upgrade" begin
-    load(joinpath(@__DIR__, "GF_2_2.json"));
-    load(joinpath(@__DIR__, "GF_2.json"));
+    loaded = load(joinpath(Main.serialization_upgrade_test_path, "version_1_2_0", "FqField-1.mrdi"))
+    @test loaded isa FqField
+    @test order(loaded) == 2
+    loaded = load(joinpath(Main.serialization_upgrade_test_path, "version_1_2_0", "FqField-2.mrdi"))
+    @test loaded isa FqField
+    @test order(loaded) == 4
     Oscar.reset_global_serializer_state()
-    load(joinpath(@__DIR__, "poly1.0.5.json"));
+    loaded = load(joinpath(Main.serialization_upgrade_test_path, "version_1_0_5", "FqMPolyRingElem.mrdi"))
```

Comment:
```suggestion
    # This is the polynomial in the "A FAIR File Format for Mathematical Software" paper
    loaded = load(joinpath(Main.serialization_upgrade_test_path, "version_1_0_5", "FqMPolyRingElem.mrdi"))
```

Just want to at least highlight this here with a comment, because it's a particularly important example.